File: docs/guides/csharp/src/searchRecentlyPublishedBooks.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchRecentlyPublishedBooks
{
  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
      var date = DateTime.UtcNow.AddYears(-1);
      var sTime = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
      var unixTime = (long)(date - sTime).TotalSeconds;
      var searchParams = new SearchParams(
        new SearchParamsObject
        {
          Query = "<YOUR_SEARCH_QUERY>",
          Filters = $"date_timestamp > {unixTime}",
        }
      );

      await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/java/src/test/java/com/algolia/searchRecentlyPublishedBooks.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;

public class searchRecentlyPublishedBooks {

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY");) {
      long dateTimestamp = ZonedDateTime.now(ZoneOffset.UTC).plusYears(-1).toEpochSecond();
      SearchParams searchParams = new SearchParamsObject().setQuery("<YOUR_SEARCH_QUERY>").setFilters("date_timestamp > " + dateTimestamp);

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}
